ng-dev/utils/ng-dev-service.ts

```diff
@@ -6,30 +6,58 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {Argv} from 'yargs';
+import {Arguments, Argv} from 'yargs';
 
-import {restoreNgTokenFromDiskIfValid} from '../auth/shared/ng-dev-token.js';
-import {initializeApp} from 'firebase/app';
+import {
+  configureAuthorizedGitClientWithTemporaryToken,
+  getCurrentUser,
+  restoreNgTokenFromDiskIfValid,
+} from '../auth/shared/ng-dev-token.js';
 import {Log} from './logging.js';
 
 /**
  * Sets up middleware to ensure that configuration and setup is completed for commands which
- *  utilize the ng-dev service
+ *  require the ng-dev service
  */
-export function requiresNgDevService(argv: Argv): Argv {
-  return argv.middleware(async () => {
-    try {
-      initializeApp({
-        apiKey: 'AIzaSyDM3rXWUgYuxYCmBKwnZvvnraYoYIE5_5U',
-        authDomain: 'internal-200822.firebaseapp.com',
-        projectId: 'internal-200822',
-        storageBucket: 'internal-200822.appspot.com',
-        messagingSenderId: '823469418460',
-        appId: '1:823469418460:web:009b51c93132b218761119',
-      });
+export function canUseNgDevService<T extends {}>(
+  argv: Argv<T>,
+  isAuthCommand: true,
+): Argv<T & {useAuthService: boolean}>;
+export function canUseNgDevService<T extends {githubToken: string | null}>(
+  argv: Argv<T>,
+): Argv<T & {useAuthService: boolean}>;
+export function canUseNgDevService<T extends {githubToken: string | null}>(
+  argv: Argv<T>,
+  isAuthCommand: boolean = false,
+): Argv<T> {
+  return argv
+    .option('use-auth-service' as 'useAuthService', {
+      type: 'boolean',
+      default: false,
+    })
+    .hide('use-auth-service')
```

Comment:
Should this flag be documented and shown?

Reply:
Added a comment, basically its just to hide it while the service is WIP